homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -303,7 +304,17 @@ async def async_step_creation(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Create config entry from external data."""
-        token = await self.flow_impl.async_resolve_external_data(self.external_data)
+        _LOGGER.debug("Creating config entry from external data")
+
+        try:
+            async with async_timeout.timeout(10):
```

Comment:
That is a very short timeout if you have some loaded network or servers in the other end that are under load. I dont think timeouts like this should be added easily.

Reply:
How.about 30?

Reply:
Better. I suppose we dont support manually cancelling a flow at the moment, so I suppose we have to have some timeout.

Reply:
Agree, 30 or even maybe 60 will do 👍

Reply:
I think 30 seconds is extremely generous for an interactive authentication flow. Updated.